src/nxt_unit.h

```diff
@@ -190,8 +185,8 @@ nxt_unit_ctx_t *nxt_unit_init(nxt_unit_init_t *);
  * from port socket should be initially processed by unit.  This function
  * may invoke other application-defined callback for message processing.
  */
-int nxt_unit_process_msg(nxt_unit_ctx_t *, nxt_unit_port_id_t *port_id,
-    void *buf, size_t buf_size, void *oob, size_t oob_size);
+int nxt_unit_process_msg(nxt_unit_ctx_t *ctx, nxt_port_msg_t *port_msg,
```

Comment:
I would like to rewrite this function as:

```
int nxt_unit_process_msg(nxt_unit_ctx_t *ctx, nxt_unit_recv_msg_t *msg, nxt_port_msg_t *port_msg)
```

But doing that requires exporting some types in `nxt_unit.h`. 
They are here: https://github.com/nginx/unit/blob/master/src/nxt_unit.c#L27

Reply:
I reverted these changes. My goal was just to handle messages in the same way as Unit does on its side:
https://github.com/nginx/unit/blob/master/src/nxt_port_socket.c#L619

But doing this requires some re-design in the unit lib.